diff --git a/pyhams/meson.build b/pyhams/meson.build
index a567087..3260519 100644
--- a/pyhams/meson.build
+++ b/pyhams/meson.build
@@ -23,13 +23,12 @@ fortranobject_c = incdir_f2py / 'fortranobject.c'
 
 inc_np = include_directories(incdir_numpy, incdir_f2py)
 
-#hams_connect = custom_target('_hamsmodule.c',
-#                             input : ['src/HAMS_Prog.f90'],
-#                             output : ['_hamsmodule.c', '_hams-f2pywrappers2.f90'],
-#                             command: [py3, '-m', 'numpy.f2py',
-#			               '@INPUT@', '-m', '_hams', '--lower',
-#                                       '--build-dir', 'pyhams']
-#                            )
+hams_connect = custom_target('_hamsmodule.c',
+                             input : ['src/HAMS_Prog.f90'],
+                             output : ['_hamsmodule.c', '_hams-f2pywrappers2.f90'],
+                             command: [py3, '-m', 'numpy.f2py',
+			               '@INPUT@', '-m', '_hams', '--lower']
+                            )
 
 # Source order is important for dependencies
 py3.extension_module('_hams',
@@ -55,11 +54,11 @@ py3.extension_module('_hams',
                      'src/SolveMotion.f90',
                      'src/WavDynSubs.f90',
                      'src/HAMS_Prog.f90',
-                     'src/_hams-f2pywrappers2.f90',
-                     'src/_hamsmodule.c',
-                     fortranobject_c,
+                     #'src/_hams-f2pywrappers2.f90',
+                     #'src/_hamsmodule.c',
+                     hams_connect,
                      ],
-                     #hams_connect,
+                     fortranobject_c,
                      include_directories: inc_np,
                      dependencies : [py3_dep, lapack, omp],
                      subdir: 'pyhams',
