diff --git a/pyhams/meson.build b/pyhams/meson.build
index a567087..074b094 100644
--- a/pyhams/meson.build
+++ b/pyhams/meson.build
@@ -57,8 +57,8 @@ py3.extension_module('_hams',
                      'src/HAMS_Prog.f90',
                      'src/_hams-f2pywrappers2.f90',
                      'src/_hamsmodule.c',
-                     fortranobject_c,
                      ],
+                     fortranobject_c,
                      #hams_connect,
                      include_directories: inc_np,
                      dependencies : [py3_dep, lapack, omp],
