diff --git a/meson.build b/meson.build
index 48f8309..c0c1105 100644
--- a/meson.build
+++ b/meson.build
@@ -15,6 +15,7 @@ project(
 cc = meson.get_compiler('c')
 add_languages('fortran', native: false)
 fc = meson.get_compiler('fortran')
+message(fc.get_id())
 is_windows = host_machine.system() == 'windows'
 
 # We need -lm for all C code (assuming it uses math functions, which is safe to
@@ -26,29 +27,57 @@ if m_dep.found()
 endif
 
 # Adding at project level causes many spurious -lgfortran flags.
+if fc.has_argument('-Wno-conversion')
+  add_project_arguments('-Wno-conversion', language: 'fortran')
+endif
+
 _fflags = []
 if fc.get_id() in ['intel', 'intel-llvm']
   _fflags = fc.get_supported_arguments('-qmkl')
+  _fflags += fc.get_supported_arguments('-fp-model=strict')
+  minus0_arg = ['-assume', 'minus0']
+  if fc.has_multi_arguments(minus0_arg)
+    _fflags += minus0_arg
+  endif
 elif fc.get_id() in ['intel-cl', 'intel-llvm-cl']
   # Intel Fortran on Windows does things differently, so deal with that
   # (also specify dynamic linking and the right name mangling)
   _fflags = fc.get_supported_arguments('/qmkl')
+  _fflags += fc.get_supported_arguments(
+    '/fp:strict', '/MD', '/names:lowercase', '/assume:underscore',
+    '/assume:minus0'
+    )  
+elif fc.get_id() in ['flang']
+  _fflags = fc.get_supported_arguments('-Xflang -Wno-everything')
 else
   _fflags = fc.get_supported_arguments('-fdec-math')
   _fflags += fc.get_supported_arguments('-fno-align-commons')
 endif
 add_project_arguments(_fflags, language: 'fortran')
 
+# Copying from scipy here
+# Platform detection
+is_mingw = is_windows and fc.get_id() == 'gcc'
+if is_mingw
+  # For mingw-w64, link statically against the UCRT.
+  gcc_link_args = ['-lucrt', '-static']
+  add_project_link_arguments('-lucrt', language: 'fortran')
+  # Flag needed to work around BLAS and LAPACK Gfortran dependence on
+  # undocumented C feature when passing single character string arguments. See:
+  #   https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90329
+  #   https://github.com/wch/r-source/blob/838f9d5a7be08f2a8c08e47bcd28756f5d0aac90/src/gnuwin32/MkRules.rules#L121
+  add_project_arguments('-fno-optimize-sibling-calls', language: 'fortran')
+endif
+
+
 omp = dependency('openmp', required: false)
 if not omp.found()
-  if fc.get_id() == 'intel'
-    omp = declare_dependency(
-      link_args: '-qopenmp',
-    )
+  if fc.get_id() in ['intel', 'intel-llvm']
+    omp = declare_dependency(link_args: '-qopenmp')
+  elif fc.get_id() in ['intel-cl', 'intel-llvm-cl']
+    omp = declare_dependency(link_args: '/qopenmp')
   else
-    omp = declare_dependency(
-      link_args: '-fopenmp',
-    )
+    omp = declare_dependency(link_args: '-fopenmp')
   endif
 endif
 
